control_panel_api/services.py

```diff
@@ -182,6 +182,27 @@ def apps3bucket_create(apps3bucket):
     )
 
 
+def apps3bucket_update(apps3bucket):
+    app_role_name = _app_role_name(apps3bucket.app.slug)
+
+    new_policy_arn = _policy_arn(
+        apps3bucket.s3bucket.name,
+        apps3bucket.has_readwrite_access(),
+    )
+    old_policy_arn = _policy_arn(
+        apps3bucket.s3bucket.name,
+        not apps3bucket.has_readwrite_access(),
+    )
+
+    aws.attach_policy_to_role(
```

Comment:
I would have thought detach needs to be called first?

Reply:
Order doesn't make much difference as long as in the end the "old" policy is detached and the "new" one is attached. The end result would be the same.